### PR TITLE
free sign up 

### PR DESCRIPTION
**Pull request by spidemen**
1. optimized the registration process
2. fix text message bug
3. fix style problem
4. fix redirect problem

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -72,10 +73,30 @@ if (process.env.NODE_ENV !== 'production') {
   require('./webpack/webpack')(app);
 }
 
+// get the base js filename
+const publicJsPath = `${__dirname}/public/js`;
+let baseJsFile;
+if (process.env.NODE_ENV === 'production') {
+  const files = fs.readdirSync(publicJsPath);
+  console.log(files);
```
Comment (spidemen) [inline app.js]: please remove this debug information

File: app.js [hunk 2]
```diff
@@ -72,10 +73,30 @@ if (process.env.NODE_ENV !== 'production') {
...
+    const stat2 = fs.statSync(`${publicJsPath}/${val2}`);
+    return stat2.mtime - stat1.mtime;
+  });
+  const reg = /^app.min.[\w]+.js$/;
+  baseJsFile = files.find(f => reg.test(f));
+}
+if (baseJsFile === undefined) {
+  baseJsFile = 'app.min.js';
+}
+logger.info(`The latest base js file is: ${baseJsFile}`);
```
Comment (spidemen) [inline app.js:92]: the same as above, if this is not a must, then remove it

File: app.js (hunk 1 above)
Comment (ety001) [reply app.js]: this has already remove on https://github.com/steemit/faucet/pull/481/commits/12a985d822f999b1151f9d13246d10038efd5cf8

File: app.js (hunk 2 above)
Comment (ety001) [reply app.js:92]: This is planned.

File: routes/apiHandlers.js [hunk 3]
```diff
@@ -1716,10 +1715,9 @@ async function handleCreateAccountNew(req) {
     });
 
     // TOS
-    const tosDate = moment().format('YYYYMMDD');
     const tosParams = [
         decoded.username,
-        `accepted_tos_${tosDate}`,
+        'accepted_tos_20180614',
```
Comment (spidemen) [inline routes/apiHandlers.js:1720]: what if we change policy, the tosDate will change too, so it is better to keep a variable like tosDate??

File: routes/apiHandlers.js [hunk 4]
```diff
@@ -1714,6 +1715,25 @@ async function handleCreateAccountNew(req) {
         }, 5 * 1000);
     });
 
+    // TOS
+    const tosDate = moment().format('YYYYMMDD');
```
Comment (spidemen) [inline routes/apiHandlers.js]: good

File: routes/apiHandlers.js (hunk 3 above)
Comment (ety001) [reply routes/apiHandlers.js:1709]: This name is a fixed name.  https://github.com/steemit/condenser/blob/master/src/server/api/general.js#L12

File: routes/apiHandlers.js (hunk 4 above)
Comment (ety001) [reply routes/apiHandlers.js]: This is an Outdated version.

**Pull request merged**